forem/forem
# Remove RssReader

## What type of PR is this? (check all applicable)


## Description

This PR rips the bandaid off removing the old `RssReader` by implementing the following things:

- removes programmatically the `:feeds_import` feature flag, via data update script
- removes `RssReader` and all related workers and classes
- renames `RssReader::Assembler` to `Feeds::AssembleArticleMarkdown`
- sets `earlier_than` to default to `nil` for `Feeds::ImportArticlesWorker` moving the default argument inside the worker to support Sidekiq Cron YAML limitations

(finally) closes https://github.com/forem/forem/issues/10996

## Related Tickets & Documents

- https://github.com/forem/forem/pull/12126
- https://github.com/orgs/forem/projects/6#card-44508281
- https://github.com/orgs/forem/projects/14#card-48867659

## QA Instructions, Screenshots, Recordings

Start the app, play with `Feeds::ImportArticlesWorker` (optionally wait for the 20th minute after the hour to make sure it's working, or change `schedule.yml` to a closer minute to verify it works :))

## Added tests?

      have not been included_

## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)

## [optional] Are there any post deployment tasks we need to perform?

We need to keep an eye on the `Articles::RssReaderWorker` job which might fail if it's picked up after deploy, no big deal in my opinion because at the following hour `Feeds::ImportArticlesWorker` will start and pick up the new work, we might just want to disable the retry manually of the former job to avoid wasting CPU cycles and resources.

We also need to remove the Heroku schedule job `fetch_all_rss` from rotation, as it's going to fail at the next cycle
